### PR TITLE
Add joystick/gamepad deadzone setting

### PR DESCRIPTION
**Pull request by maromalo**
- [x] Depends on ppy/osu-framework#5122
- [x] Depends on framework bump

Adds a joystick subsection and deadzone configuration.

### CODE REVIEW

**Comment by peppy** [comment]
Please make sure to read the contribution guidelines and use a non-`master` branch in the future.

File: osu.Game/Localisation/JoystickSettingsStrings.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        /// <summary>
+        /// "Joystick / Gamepad"
+        /// </summary>
+        public static LocalisableString JoystickGamepad => new TranslatableString(getKey(@"joystick_gamepad"), @"Joystick / Gamepad");
+
+        /// <summary>
+        /// "Deadzone Threshold"
+        /// </summary>
+        public static LocalisableString DeadzoneThreshold => new TranslatableString(getKey(@"deadzone_threshold"), @"Deadzone Threshold");
```
Comment (frenzibyte) [inline osu.Game/Localisation/JoystickSettingsStrings.cs:20]: "Deadzone" and "Threshold" definitely don't go together. Would've called this "Analog stick deadzone" but it seems to apply to triggers as well. Just "Deadzone" should suffice.
```suggestion
        public static LocalisableString DeadzoneThreshold => new TranslatableString(getKey(@"deadzone_threshold"), @"Deadzone");
```

File: osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            enabled.BindTo(joystickHandler.Enabled);
+            deadzoneThreshold.BindTo(joystickHandler.DeadzoneThreshold);
+            enabled.BindValueChanged(e => deadzoneThreshold.Disabled = !e.NewValue, true);
+        }
+
+        private class DeadzoneSetting : SettingsSlider<float, DeadzoneSlider>
+        {
+            public DeadzoneSetting()
+            {
+                KeyboardStep = 0.005f;
```
Comment (bdach) [inline osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs:56]: Not really sure why this step is so tiny. Does anyone really need this fine-grained control over the deadzone? I'd say the step could be 10x what it is right now and nobody should have a problem with that...

**Review by bdach — COMMENTED** [review]
Seems to mostly work in testing, just one doubt re: how the setting is specified

File: osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs (hunk 2 above)
Comment (maromalo) [reply osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs:56]: Osu's default deadzone was `0.075f`, I figured a step of 0.005 would make that values like that easier to get to, though I can consider 0.01 (about 100 choices from 0 to 0.95). A step bigger than that seems way too big of a skip imo.

Comment (bdach) [reply osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs:56]: I could live with 0.01 too I guess.

**Review by bdach — DISMISSED** [review]

File: osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            public DeadzoneSetting()
+            {
+                KeyboardStep = 0.01f;
+                TransferValueOnCommit = true;
+            }
+        }
+
+        private class DeadzoneSlider : OsuSliderBar<float>
+        {
+            public override LocalisableString TooltipText => Current.Disabled ? "" : base.TooltipText;
```
Comment (peppy) [inline osu.Game/Overlays/Settings/Sections/Input/JoystickSettings.cs:63]: I'm not sure I see the point of this. Seems like something that should be done at a lower level?

**Review by peppy — APPROVED** [review]

**Pull request merged**